src/Integration.UnitTests/Binding/CSharpVBBindingOperationTests.cs

```diff
@@ -396,13 +396,74 @@ public void ProjectBindingOperation_Prepare_NoAdditionalFileConflict_NoException
             additionalFileConflictChecker.VerifyAll();
         }
 
+        [TestMethod]
+        public void ProjectBindingOperation_Commit_AllRuleSetsAreDefault_AddsNonConditionalRuleSetProperty()
+        {
+            // Arrange
+            projectMock.SetCSProjectKind();
+            var testSubject = CreateTestSubject();
+            var debug = CreateRuleSetProperty(projectMock, "Debug", CSharpVBBindingOperation.DefaultProjectRuleSet);
+            var release = CreateRuleSetProperty(projectMock, "Release", CSharpVBBindingOperation.DefaultProjectRuleSet);
+
+            testSubject.Initialize();
+            testSubject.Prepare(CancellationToken.None);
+
+            var nonConditionalProperty = projectMock.GetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+            nonConditionalProperty.Should().BeNullOrEmpty();
+
+            // Act
+            using (new AssertIgnoreScope()) // Ignore that the file is not on disk
+            {
+                testSubject.Commit();
+            }
+
+            // Assert
+            var solutionLevelRuleSet = PathHelper.CalculateRelativePath(testSubject.ProjectFullPath, cSharpVBBindingConfig.RuleSet.Path);
+
+            nonConditionalProperty = projectMock.GetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+            nonConditionalProperty.Should().Be(solutionLevelRuleSet);
+            debug.Value.ToString().Should().Be(solutionLevelRuleSet);
+            release.Value.ToString().Should().Be(solutionLevelRuleSet);
+        }
+
+        [TestMethod]
+        public void ProjectBindingOperation_Commit_HasNonDefaultRuleSet_DoesNotAddNonConditionalRuleSetProperty()
+        {
+            // Arrange
+            projectMock.SetCSProjectKind();
+
+            var testSubject = CreateTestSubject();
+            var defaultRuleSet = CreateRuleSetProperty(projectMock, "Debug", CSharpVBBindingOperation.DefaultProjectRuleSet);
+            var nonDefaultRuleSet = CreateRuleSetProperty(projectMock, "Release", "myruleset.ruleset");
+
+            testSubject.Initialize();
+            testSubject.Prepare(CancellationToken.None);
+
+            // Act
+            using (new AssertIgnoreScope()) // Ignore that the file is not on disk
+            {
+                testSubject.Commit();
+            }
+
+            // Assert
+
+            var nonConditionalProperty = projectMock.GetBuildProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+            nonConditionalProperty.Should().BeNullOrEmpty();
+
+            var solutionLevelRuleSet = PathHelper.CalculateRelativePath(testSubject.ProjectFullPath, cSharpVBBindingConfig.RuleSet.Path);
```

Comment:
Minor - `generatedRuleSet` is more accurate than `solutionLevel` - it's clearer that it means "our" ruleset.
It's possible to have a solution-level ruleset that is not the generated ruleset, and we probably have some tests that have such a configuration.